app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HabitViewerViewModel.kt

```
@@ -121,6 +151,82 @@ class HabitViewerViewModel @Inject constructor(
+        val tempFile = File.createTempFile("habit-", extension, appContext.cacheDir)
+        return@withContext try {
+            val bytes = Base64.decode(base64, Base64.DEFAULT)
+            tempFile.writeBytes(bytes)
+            when (val result = uploadRepo.upload(tempFile.path)) {
+                is ApiResult.Ok -> sanitizeImageUrl(result.data.url ?: result.data.path)
+                is ApiResult.Err -> {
+                    Clogger.e("HabitViewerViewModel", "Image upload failed: ${result.message}")
+                    null
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)